### DISCUSSION TITLE
A CNAME domain doesn't appear as CNAME in the UI popup after some time

### DISCUSSION THREAD

**Discussion by uBlock-user** [uBlockOrigin/uBlock-issues#3376] [General]
STR -- 

- Browse to `https://putlocker.pe/watch-tv/watch-the-lord-of-the-rings-the-rings-of-power-online-87087.10655554`

- Watch the episode for some time OR switch episodes and wait for some time again then and open uBlock UI popup by clicking on the extension icon

`jwplayer-dualstack.map.fastly.net` which is a CNAME of `ssl.p.jwpcdn.com` doesn't appear in blue color as CNAME but rather any ordinary domain and in white color in the popup UI.

Actual  - ![Capture](https://github.com/user-attachments/assets/aa8919c9-d15a-4ecc-ad9e-48fef1ecf512)

Expected - ![Capture2](https://github.com/user-attachments/assets/350d98de-ddbd-4984-b96e-d8d50b67ba22)

Q : IS this a regression from the CNAME rewrite commit ?

@gorhill

**Comment by gorhill** [comment]
Version of uBO?

**Reply by uBlock-user** [reply]
1.59.1b21 dev build

**Reply by gorhill** [reply]
Alright, the issue was always there, it just now became more visible because the dns records are flushed from the cache after 60 seconds instead of 2 hours. Have to give thoughts about how this can be fixed.

**Reply by uBlock-user** [reply]
But how come only this domain is seen visibly affected ? I didn't see any other CNAME get affected like this.

**Reply by gorhill** [reply]
They are affected, just reload the page, take note of all the uncloaked hostnames, wait > 61 seconds, see all of them no longer being reported as result of uncloaking.

**Reply by uBlock-user** [reply]
https://github.com/uBlockOrigin/uBlock-issues/issues/3377

**Reply by gorhill** [reply]
Fixed in <https://github.com/gorhill/uBlock/commit/c265e849e070edf49a73f14007bf35f2a8a34656>, I didn't see an issue was created for this before committing.

**Reply by uBlock-user** [reply]
How come the CNAME entries in the logger don't get affected ?

**Reply by gorhill** [reply]
Because in the logger the cname value is not dynamically fetched, it's part of the logged information. They were dynamically fetched for the hostname database part of the page details. No longer the case, they are fetched and stored along other hostname details such as the count of requests per hostname.

**Reply by uBlock-user** [reply]
Happening again. Today happened at random, you can try here - `https://putlocker.pe/watch-tv/watch-stargate-sg1-online-39410.2431543`

second url - `https://www.reddit.com/r/scifi/comments/1fm39h2/missing_this_i_always_thought_it_was_such_a_great/`

uBO version - 1.59.1rc6

@gorhill can you reproduce ?

**Reply by gorhill** [reply]
What's the steps to reproduce? For the previously reported issue, the steps were load the page and wait 61s before bringing up the popup panel. What is it now? Need to navigate? Wait? If so how long?

**Reply by uBlock-user** [reply]
I don't have any steps. That's why I said it happens out of random. Previous steps certainly don't work as you added a fix for that. 

For navigation I have posted two url where I encountered that issue again.

**Reply by gorhill** [reply]
I think there is a small time window in which the cached cname value may go away after the corresponding request has been handled by the listener and before its recorded in a page's internal details, and that time window would correspond to the value of advanced setting [`requestJournalProcessPeriod`](https://github.com/gorhill/uBlock/wiki/Advanced-settings#requestjournalprocessperiod), which default to 1 second.

**Reply by uBlock-user** [reply]
> to the value of advanced setting [requestJournalProcessPeriod](https://github.com/gorhill/uBlock/wiki/Advanced-settings#requestjournalprocessperiod),

So if the value is < `1000`, this issue will happen ?

**Reply by gorhill** [reply]
The higher the value, the more likely it should happen.

**Reply by uBlock-user** [reply]
Mine is set to `500`, so I have now set it to `400`, lets see if that helps.

**Reply by gorhill** [reply]
If the issue is what I think it is, should be fixed with <https://github.com/gorhill/uBlock/commit/e7c783cefaea69e709281b02dd1efb970da9e412>.

**Reply by uBlock-user** [reply]
I can reproduce the issue here -- `https://anime8.ru/Anime/Chained-Soldier.58882/Episode-010?id=189305` intermittently

You need to setup hardmode mode like i have with these rules --

```
no-csp-reports: * true
* * 3p block
* * 3p-frame block
* * 3p-script block
anime8.ru akamai.net * noop
anime8.ru bytedance.map.fastly.net * noop
anime8.ru ibytedtos.com * noop
anime8.ru ssl.p.jwpcdn.com * noop
anime8.ru jwplayer-dualstack.map.fastly.net * noop
anime8.ru cdn.jwplayer.com * noop
anime8.ru openstream.io * noop
anime8.ru rsc.cdn77.org * noop
anime8.ru embed.vodstream.xyz * noop
```

- Navigate to the page above and click on the play button to play the video and now open the popup panel and you should notice `*.akamai.net` being lodged in white colour. You may need to try this step many times if it happens intermittently like on my end.

version 1.60.1b1

`requestJournalProcessPeriod` is set to `500`


This is how it looks like, sometimes all are white, sometimes some blue some white.
![Capture](https://github.com/user-attachments/assets/8a91b616-245b-425a-8c50-507b35f431fd)
